media-capabilities/index.js

```diff
@@ -13,25 +13,27 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+import { useState, useEffect } from 'react';
 
 const supported = typeof window !== 'undefined' && 'mediaCapabilities' in navigator;
 
-const useMediaCapabilities = (mediaConfig, initialMediaCapabilities = {}) => {
-  let mediaCapabilities = {
-    supported,
-    hasMediaConfig: !!mediaConfig
+const useMediaCapabilitiesDecodingInfo = (mediaDecodingConfig, initialMediaCapabilitiesInfo = {}) => {
+  initialMediaCapabilitiesInfo = {
+    ...initialMediaCapabilitiesInfo
   };
 
-  mediaCapabilities = (mediaCapabilities.supported && mediaCapabilities.hasMediaConfig)
-    ? navigator.mediaCapabilities.decodingInfo(mediaConfig)
-    : {
-      ...mediaCapabilities,
-      ...initialMediaCapabilities
-    };
+  const [mediaCapabilitiesInfo, setMediaCapabilitiesInfo] = useState(initialMediaCapabilitiesInfo);
 
-  return {mediaCapabilities};
-};
+  useEffect(() => {
+    supported &&
+    navigator
+      .mediaCapabilities
+      .decodingInfo(mediaDecodingConfig)
+      .then(setMediaCapabilitiesInfo)
+      .catch(error => console.error(error));
+  })
```

Comment:
@wingleung 

I wonder if this raises hook warning.
As well, I think the logic inside `useEffect` hook would be executed every re-rendering of the component with this hook. It could not be good for performance.

I think it could be something like the following.
```
useEffect(() => {
    supported &&
    navigator
      .mediaCapabilities
      .decodingInfo(mediaDecodingConfig)
      .then(setMediaCapabilitiesInfo)
      .catch(error => console.error(error));
  }, [supported, mediaDecodingConfig]);
```

Reply:
We can pass an empty array to have the `componentDidMount` effect.

> If you want to run an effect and clean it up only once (on mount and unmount), you can pass an empty array ([]) as a second argument. 
https://reactjs.org/docs/hooks-faq.html#what-can-i-do-if-my-effect-dependencies-change-too-often

Reply:
Thank you @wingleung for your updating.

Yes, you're correct.
As far as I know, if then, the useEffect hook would raise warnings because dependencies like `supported` and `mediaDecodingConfig` are missing.

https://stackoverflow.com/questions/55840294/how-to-fix-missing-dependency-warning-when-using-useeffect-react-hook

If we tried that hook in the real react project (dev env), then we could know it.

Reply:
If possible, would you mind adding ; at the end of useEffect hook for consistency.